### PR TITLE
[INTERNAL] Use GitHub Actions for Coveralls reporting instead of Travis CI

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

File: .github/workflows/coveralls.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on: ["push", "pull_request"]
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+    - uses: actions/checkout@v1
```
Comment (matz3) [inline .github/workflows/coveralls.yml:12]: There's a v2
```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/coveralls.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      with:
+        node-version: 14.x
+
+    - name: Install dependencies and execute tests
+      run: |
+        npm ci
+        npm run coverage
+
+    - name: Coveralls
+      uses: coverallsapp/github-action@master
```
Comment (matz3) [inline .github/workflows/coveralls.yml:25]: I'd prefer to use a fixed version
```suggestion
      uses: coverallsapp/github-action@v1.1.2
```

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-
-node_js:
-- "lts/*"
-
-script: npm run coverage
-
-after_script:
-- if [[ "$TRAVIS_OS_NAME" == "linux" &&
-    (( -n "$TRAVIS_PULL_REQUEST" && "$TRAVIS_PULL_REQUEST" != "false" ) || "$TRAVIS_BRANCH" == "master" ) ]]; then
-    npm run report-coveralls;
```
Comment (matz3) [inline .travis.yml:11]: We can now remove the `report-coveralls` npm script, right?

File: .github/workflows/coveralls.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+name: Coveralls
+
+on: ["push", "pull_request"]
```
Comment (matz3) [inline .github/workflows/coveralls.yml:3]: We should only do this for PRs and pushes to master. That's also how it has been done in Travis.

```suggestion
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
```

**Review by matz3 — CHANGES_REQUESTED** [review]

File: .github/workflows/coveralls.yml (hunk 1 above)
Comment (RandomByte) [reply .github/workflows/coveralls.yml]: Done

File: .travis.yml (hunk 3 above)
Comment (RandomByte) [reply .travis.yml:11]: Done

File: .github/workflows/coveralls.yml (hunk 4 above)
Comment (RandomByte) [reply .github/workflows/coveralls.yml]: Done

File: .github/workflows/coveralls.yml (hunk 2 above)
Comment (RandomByte) [reply .github/workflows/coveralls.yml]: Done

**Review by matz3 — APPROVED** [review]

**Pull request merged**